debug-helpers/debug-helper.js

```diff
@@ -0,0 +1,116 @@
+
+window.onload = (event) => {
+  refreshVariables();
+};
+
+var map = new mapboxgl.Map({
+  container: "map",
+  style: "mapbox://styles/mapbox/light-v9",
+  center: [-121.893028, 37.33548], // position in long, lat format
+  zoom: 10,
+  dragPan: true, // If true , the "drag to pan" interaction is enabled (see DragPanHandler)
+  trackResize: true, // If true, the map will automatically resize when the browser window resizes.
+  doubleClickZoom: true, //If true double click will zoom
+  keyboard: true //If true will enable keyboard shortcuts
+});
+
+map.on("load", function(e) {
+  if (!latitude || !longitude) {
+    // missing coordinates
+    console.log("artwork has missing coordinates")
+  } else {
+    // coordinates are present
+    mapFlyToCurrent();
+  }
+});
+
+function queryNominatim() {
+  refreshVariables();
+  var nominatimAPI = `https://nominatim.openstreetmap.org/?`;
+  var query = { 
+    params:{
+      q: address + ", " + city + ", " + state + ", " + postalCode,
+      format: "geocodejson",
+      limit: "1",
+    }
+  }
+  axios
+    .get(nominatimAPI, query)
+    .then(
+      function(response) {
+        [longitude, latitude] = response.data.features[0].geometry.coordinates;
+        document.getElementById("longitude").value = longitude;
+        document.getElementById("latitude").value = latitude; 
+        refreshVariables();
+        mapFlyToCurrent();
+      })
+    .catch(
+      function(error) {
+        alert("Error getting coordinates from Nominatim. Please try a different address.")
+    });
+}
+
+function submitPOI() {
+  refreshVariables();
+  if (!latitude || !longitude) {
+    alert("Missing Coordinates. Cannot Submit");
+    return
+  }
+  if (!title || !artist || !description) {
+    alert("Missing Details. Cannot Submit")
+    return
+  }
+
+  var pathTo = "/POI";
+  if (pageType === "Edit") {pathTo = `/POI/${id}`}
+  axios.post(pathTo, {
+    title, artist, address, description, city, state, postalCode, image, sourceURL, id, latitude, longitude
+  })
+  .then( response => {
+    const data = response.data
+    alert(`Successfully submitted artwork`);
+    window.location = "/";
+  }).catch(error => {
+    alert(error)
+  })
+}
+
+function mapFlyToCurrent() {
+  refreshVariables();
+  if (!latitude || !longitude) { console.log("Unable to fly to artwork. Missing coordinates"); return }
+  map.flyTo({
+    center: [longitude, latitude],
+    zoom: 15
+  });
+  if (popup) {popup.remove()}
+  var popup = new mapboxgl.Popup({ closeOnClick: true, closeButton: true, anchor: "top" })
+  .setLngLat([longitude, latitude])
+  .setHTML(
+  `<div style="width: 300px;">
+    <h3>${title}</h3>
+    <p>by ${artist}</p>
+    <p>Address: ${address}, ${city}, ${state}, ${postalCode}</p>
+    <div>
+      <img style="width: 300px" src=${image}>
+    </div>
+    <p>Description: ${description}</p>
+  </div>`
+  ).addTo(map);
+}
+
+function refreshVariables() {
+  title = document.getElementById("title").value;
```

Comment:
Ideally this should be mapped values in a loop, and each element should be checked to see if it exists before just pulling the value.

Reply:
noted on this